nndownload/nndownload.py

```diff
@@ -1286,29 +1309,17 @@ def download_video_media(session: requests.Session, filename: AnyStr, template_p
         # .part file
         if os.path.exists(filename):
             output("Resuming partial downloads is not supported for videos using DMS delivery. Any partial video data will be overwritten.\n", logging.WARNING)
-        if _cmdl_opts.threads:
-            output("Multithreading is only supported for DMC delivery. Video will be downloaded using one thread.\n", logging.WARNING)
 
         m3u8_streams = []
-        with get_temp_dir() as temp_dir:
-            for stream_type in ["dms_video_uri", "dms_audio_uri"]:
-                if template_params.get(stream_type):
-                    m3u8_path = os.path.join(temp_dir, f"{template_params['id']}_{stream_type}.m3u8")
-                    m3u8 = generic_dl_request(session, template_params[stream_type], m3u8_path)
-                    # It's minimally viable to only rewrite the key file locally for now
-                    # Might be wise to eventually do this with the map and all individual segments
-                    key_match = M3U8_KEY_RE.search(m3u8)
-                    if not key_match:
-                        raise FormatNotAvailableException("Could not retrieve key file from manifest")
-                    key_url = key_match[2]
-                    key_path =  os.path.join(temp_dir, f"{template_params['id']}_{stream_type}.key")
-                    key_path = key_path.replace("\\", "/")
-                    generic_dl_request(session, key_url, key_path, binary=True)
-                    rewrite_file(m3u8_path, key_url, key_path)
-                    m3u8_streams.append(m3u8_path)
-            continue_code = perform_ffmpeg_dl(template_params["id"], filename, float(template_params["duration"]), m3u8_streams)
-            os.rename(filename, complete_filename)
-            return continue_code
+        for stream_type, type_ext in [("dms_video_uri", "vid"), ("dms_audio_uri", "aud")]:
+            if template_params.get(stream_type):
+                stream_filename = filename + "." + type_ext
```

Comment:
I'd like us to keep using `get_temp_dir()` here, we shouldn't need to rewrite the extensions or expose the streams on the filesystem

Reply:
I can change it to use get_temp_dir().

My only concern is that compared to just saving meta files like `.m3u8` and key there, these separate stream files would be relatively large, and may cause unexpected disk usage on their main drive (e.g. on Windows, get_temp_dir() would just use %TEMP% which is `C:\Users\{username}\AppData\Local\Temp`) rather than their CWD. And it seems common for downloader programs to use CWD for their partial downloaded files (yt-dlp, aria2c, to name a few).

But I guess we should be fine.